README.md

```diff
@@ -2,157 +2,317 @@
 
 ## Structure
 
-* [Demo](#demo)
-* [Get started](#getting-started)
-    + [Prerequisites](#prerequisites)
-    + [Installation](#installation)
-    + [Usage](#usage)
-        + [Configurations](#configurations)
-            + [TOML files](#toml-files)
-                + [Global keys](#global-keys)
-                + [Custom Prelude Options](#custom-prelude-options)
-                + [TOML file Examples](#examples)
-            + [CLI](#cli)
-* [Features](#features)
-* [GHC options](#ghc-options)
-* [Change log](#change-log)
-* [Acknowledgments](#acknowledgments)
-
-## Demo
+* [Demo](#demo-)
+    + [TUI demo](#tui-demo-)
+    + [CLI demo](#cli-demo-)
+    + [Scaffolded project structure](#scaffolded-project-structure-)
+* [Features](#features-)
+* [Get started](#get-started-)
+    + [Prerequisites](#prerequisites-)
+    + [Installation](#installation-)
+        + [Summon-TUI](#summon-tui-)
+          + [TUI: from source](#tui-from-source-)
+        + [Summon-CLI](#summon-cli-)
+          + [CLI: from Hackage/Stackage](#cli-from-hackagestackage-)
+          + [CLI: from source](#cli-from-source-)
+    + [Setup](#setup-)
+* [Usage](#usage-)
+    + [Command-line arguments](#command-line-arguments-)
+    + [File configuration](#file-configuration-)
+    + [TUI](#tui-)
+    + [CLI](#cli-)
+* [For Developers](#for-developers-)
+    + [Build](#build-)
+    + [Test](#test-)
+    + [Run](#run-)
+* [Changelog](#changelog-)
+* [Acknowledgments](#acknowledgments-)
+
+## Demo [↑](#structure)
+
+### TUI demo [↑](#structure)
+
+TODO: TUI demo asciinema
+
+### CLI demo [↑](#structure)
 
 [![asciicast](https://asciinema.org/a/198918.png)](https://asciinema.org/a/198918)
 
-## Getting started
+### Scaffolded project structure [↑](#structure)
 
-### Prerequisites
+This is the example of the project hierarchy you can get if you're running Summoner with all options enabled:
 
-To start using it make sure you have the next tools installed on your machine:
+```
+project-name/
+├── app/
+│   └── Main.hs
+├── benchmark/
+│   └── Main.hs
+├── src/
+│   ├── ProjectName.hs
+│   └── Prelude.hs
+├── test/
+│   └── Spec.hs
+├── CHANGELOG.md
+├── CONTRIBUTING.md
+├── LICENSE
+├── project-name.cabal
+├── README.md
+├── Setup.hs
+├── stack.yaml
+├── appveyor.yml
+├── .git
+├── .gitignore
+├── .stylish-haskell.yaml
+└── .travis.yml
+```
 
-* [`Stack`](http://haskellstack.org) or [`cabal`](https://www.haskell.org/cabal/)
-* [`git`](https://git-scm.com)
-* [`hub`](https://github.com/github/hub)
-* [`curl`](https://curl.haxx.se)
+You can also see complete examples in the following folder:
 
-### Installation
+* [`summon-cli/tests/golden`: Examples of scaffolded projects](https://github.com/kowainik/summoner/tree/master/summoner-cli/test/golden)
 
-The installation process can be done with the one simple command:
+Directory `smallProject` contains an example of the straightforward package, and `fullProject` shows generated project with more advanced structure and more features.
```

Comment:
`fullProject` shows *a* generated project with a more advanced structure and more features.